test/unit/puppetlabs/puppetserver/ruby/http_client_test.clj

```diff
@@ -215,25 +235,44 @@
                    :cipher-suites ["SSL_RSA_WITH_RC4_128_SHA"]})]
           (.runScriptlet sc "response = c.get('/', {})")
           (is (= "200" (.runScriptlet sc "response.code")))
-          (is (= "hi" (.runScriptlet sc "response.body"))))))))
+          (is (= "hi" (.runScriptlet sc "response.body")))
+          (terminate-scripting-container sc))))))
 
 (deftest clients-persist
   (testing "client persists when making HTTP requests"
     (logutils/with-test-logging
       (jetty9/with-test-webserver ring-app port
-        (let [scripting-container (create-scripting-container port)
-              client1 (.runScriptlet scripting-container "c.get('/', {}); c.client")
-              client2 (.runScriptlet scripting-container "c.post('/', 'foo', {}); c.client")]
-          (is (= client1 client2)))))))
+        (let [sc (create-scripting-container port)
+              client1 (.runScriptlet sc "c.get('/', {}); c.class.client")
+              client2 (.runScriptlet sc "c.post('/', 'foo', {}); c.class.client")]
+          (is (= client1 client2))
+          (terminate-scripting-container sc))))))
```

Comment:
Since we missed this the first time, I think it would be good to create a second test here where we initialize another `Puppet::Server::HttpClient` instance and validate that its ".class.client" is identical to the ".class.client" from the original `Puppet::Server::HttpClient` instance.